pkg/operator/controller/externaldns/pod.go

```diff
@@ -93,19 +94,19 @@ func newExternalDNSContainerBuilder(image, provider, source string, secret *core
 }
 
 // build returns the definition of a single container for the given DNS zone with unique metrics port
-func (b *externalDNSContainerBuilder) build(zone string) *corev1.Container {
+func (b *externalDNSContainerBuilder) build(zone string) (*corev1.Container, error) {
```

Comment:
I don't know how much it's useful to specify both `zone-id-filter` and `domain-filter` but our API and external-dns allow this, so should we I think.

However I think the other use case is more likely to be used for the domains: no `zone-id-filter`, only domain filters. And this use case needs to be covered too.

For the moment, the composition of the desired deployment is built around the zones (1 container per zone). I think that we need to add the special case when no zone is specified but there are domains.

Reply:
IIUC they should not interfere with each other. I think when there is no zone specified then `external-dns` will attempt to reconcile all potential resources. And will filter on the `domain-filter` values if specified.

Reply:
Yes, they don't interfere. And yes, if only domain filter is specified external-dns will list all hosted zones and filter on the given domain.

The problem is that if no `.Spec.Zones` was set in CR the external-dns deployment will not have a single container.

Reply:
I can write a card for that case. Let's address that in a separate PR.

Reply:
Xref the card: [CFE-142](https://issues.redhat.com/browse/CFE-142).